core/schema/host.go

```diff
@@ -5,60 +5,102 @@ import (
 	"errors"
 	"fmt"
 
+	"github.com/containerd/containerd/labels"
 	"github.com/dagger/dagger/core"
-	"github.com/dagger/dagger/core/socket"
+	"github.com/dagger/dagger/dagql"
+	"github.com/dagger/dagger/engine/sources/blob"
+	"github.com/opencontainers/go-digest"
+	specs "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
 type hostSchema struct {
-	*APIServer
-
-	host *core.Host
-	svcs *core.Services
+	srv *dagql.Server
 }
 
 var _ SchemaResolvers = &hostSchema{}
 
-func (s *hostSchema) Name() string {
-	return "host"
-}
-
-func (s *hostSchema) Schema() string {
-	return Host
-}
-
-func (s *hostSchema) Resolvers() Resolvers {
-	return Resolvers{
-		"Query": ObjectResolver{
-			"host": PassthroughResolver,
-		},
-		"Host": ObjectResolver{
-			"directory":     ToResolver(s.directory),
-			"file":          ToResolver(s.file),
-			"unixSocket":    ToResolver(s.socket),
-			"setSecretFile": ToResolver(s.setSecretFile),
-			"tunnel":        ToResolver(s.tunnel),
-			"service":       ToResolver(s.service),
-		},
-	}
+func (s *hostSchema) Install() {
+	dagql.Fields[*core.Query]{
+		dagql.Func("host", func(ctx context.Context, parent *core.Query, args struct{}) (*core.Host, error) {
+			return parent.NewHost(), nil
+		}).Doc(`Queries the host environment.`),
+		dagql.Func("blob", func(ctx context.Context, parent *core.Query, args struct {
+			Digest       string `doc:"Digest of the blob"`
+			Size         int64  `doc:"Size of the blob"`
+			MediaType    string `doc:"Media type of the blob"`
+			Uncompressed string `doc:"Digest of the uncompressed blob"`
+		}) (*core.Directory, error) {
+			blobDef, err := blob.LLB(specs.Descriptor{
+				MediaType: args.MediaType,
+				Digest:    digest.Digest(args.Digest),
```

Comment:
nit: `digest.Parse` will verify that the string is actually the right format, probably worth it here (though this is essentially only used internally atm so not a big deal). Can do the same for `args.Uncompressed`.

Reply:
Sounds good. Tempting to add a `Digest` scalar type. :thinking: I'll punt on that for now though, can't think of anywhere else we'd use it yet.